src/main/java/org/jenkinsci/plugins/docker/commons/fingerprint/DockerFingerprints.java

```diff
@@ -66,7 +66,12 @@ public static void addRunFacet(@Nonnull ContainerRecord record, @Nonnull Run<?,?
     /**
      * Creates a new {@link DockerFromFingerprintFacet} and adds a run. Or adds to an existing one.
      */
-    public static void addFromFacet(@Nonnull String imageId, @Nonnull Run<?,?> run) throws IOException {
+    public static void addFromFacet(@Nonnull String baseImageId, @Nonnull String derivedImageId, @Nonnull Run<?,?> run) throws IOException {
+        long timestamp = System.currentTimeMillis();
+        addHalfFromFacet(baseImageId, derivedImageId, false, run, timestamp);
+        addHalfFromFacet(derivedImageId, baseImageId, true, run, timestamp);
+    }
+    private static void addHalfFromFacet(@Nonnull String imageId, @Nonnull String otherImageId, boolean inverse, @Nonnull Run<?,?> run, long timestamp) throws IOException {
```

Comment:
needs some comments on what this is doing and why because at first glance it's not obvious.